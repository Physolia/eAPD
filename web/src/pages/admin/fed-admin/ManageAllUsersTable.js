/* eslint-disable react/prop-types */
import PropTypes from 'prop-types';
import React, { Fragment } from 'react';

import {
  Table,
  TableHead,
  TableRow,
  TableCell,
  TableBody
} from '@cmsgov/design-system';

const ManageAllUsersTable = ({
  tab,
  affiliations,
  isFetching,
  actions,
  currentUser
}) => {
  const { id: currentUserId, activities: currentUserActivities } = currentUser;

  const showActions = affiliation => {
    return (
      currentUserId !== affiliation.userId &&
      currentUserActivities.indexOf('edit-affiliations') !== -1
    );
  };

  /* eslint-disable react/no-unstable-nested-components */
  const AffiliationFirstRow = ({ primaryAffiliation, affiliation }) => {
    return (
      <TableRow
        className={
          primaryAffiliation.affiliations.length > 1
            ? 'all-users-table--first-row-multi'
            : ''
        }
        key={affiliation.id}
      >
        <TableCell
          component="th"
          style={{ verticalAlign: 'top' }}
          rowSpan={primaryAffiliation.affiliations.length}
        >
          {primaryAffiliation.displayName}
        </TableCell>
        <TableCell
          component="td"
          style={{ verticalAlign: 'top' }}
          rowSpan={primaryAffiliation.affiliations.length}
        >
          {primaryAffiliation.email}
        </TableCell>
        <TableCell
          component="td"
          style={{ verticalAlign: 'top' }}
          rowSpan={primaryAffiliation.affiliations.length}
        >
          {primaryAffiliation.primaryPhone}
        </TableCell>
        <TableCell className="all-users-table--state">
          {affiliation.stateId.toUpperCase()}
        </TableCell>
        {tab === 'active' ? (
          <TableCell className="all-users-table--role">
            {affiliation.role}
          </TableCell>
        ) : null}
        {tab === 'inactive' ? (
          <TableCell className="all-users-table--status">
            {affiliation.status}
          </TableCell>
        ) : null}
        <TableCell className="all-users-table--actions">
          <AffiliationActions
            primaryAffiliation={primaryAffiliation}
            affiliation={affiliation}
          />
        </TableCell>
      </TableRow>
    );
  };
  const AffiliationRow = ({ primaryAffiliation, affiliation }) => {
    return (
      <TableRow
        className={
          primaryAffiliation.affiliations.length > 1
            ? 'all-users-table--row-multi'
            : ''
        }
        key={affiliation.id}
      >
        <TableCell className="all-users-table--state">
          {affiliation.stateId.toUpperCase()}
        </TableCell>
        {tab === 'active' ? (
          <TableCell className="all-users-table--role">
            {affiliation.role}
          </TableCell>
        ) : null}
        {tab === 'inactive' ? (
          <TableCell className="all-users-table--status">
            {affiliation.status}
          </TableCell>
        ) : null}
        <TableCell className="all-users-table--actions">
          <AffiliationActions
            primaryAffiliation={primaryAffiliation}
            affiliation={affiliation}
          />
        </TableCell>
      </TableRow>
    );
  };

  const AffiliationActions = ({ primaryAffiliation, affiliation }) => {
    // Only show revoked for the active tab
    let displayedActions;
    if (tab !== 'active') {
      displayedActions = [...actions];
    }
    if (tab === 'active') {
      displayedActions = [actions[1]];
    }
    return (
      <div
        className="ds-u-display--flex"
        data-primary-affiliation-id={primaryAffiliation.id}
        data-id={affiliation.id}
        data-state={affiliation.stateId}
      >
        {showActions(primaryAffiliation) && displayedActions}
      </div>
    );
  };

  return (
    <Fragment>
      {isFetching && <p>Loading...</p>}
      {!isFetching && affiliations.length === 0 && (
        <p>No users on this tab at this time</p>
      )}
      {!isFetching && affiliations.length > 0 && (
        <Table borderless className="all-users-table">
          <TableHead>
            <TableRow>
              <TableCell>Name</TableCell>
              <TableCell>Email</TableCell>
              <TableCell>Phone Number</TableCell>
              <TableCell>State</TableCell>
              {tab === 'active' ? <TableCell>Role</TableCell> : null}
              {tab === 'inactive' ? <TableCell>Status</TableCell> : null}
              <TableCell>Actions</TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            {affiliations.map(primaryAffiliation => (
              <Fragment key={primaryAffiliation.id}>
                {primaryAffiliation.affiliations.map((affiliation, index) => {
                  if (index === 0) {
                    return (
                      <AffiliationFirstRow
                        key={affiliation.id}
                        primaryAffiliation={primaryAffiliation}
                        affiliation={affiliation}
                      />
                    );
                  }
                  return (
                    <AffiliationRow
                      key={affiliation.id}
                      primaryAffiliation={primaryAffiliation}
                      affiliation={affiliation}
                    />
                  );
                })}
              </Fragment>
            ))}
          </TableBody>
        </Table>
      )}
    </Fragment>
  );
};

ManageAllUsersTable.propTypes = {
  tab: PropTypes.string.isRequired,
  affiliations: PropTypes.array.isRequired,
  isFetching: PropTypes.bool.isRequired,
  actions: PropTypes.array,
  currentUser: PropTypes.object.isRequired
};

ManageAllUsersTable.defaultProps = {
  actions: ''
};

export default ManageAllUsersTable;
