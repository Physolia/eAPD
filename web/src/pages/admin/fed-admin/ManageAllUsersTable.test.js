import React from 'react';
import { render } from 'apd-testing-library';
import ManageAllUsersTable from './ManageAllUsersTable';

let props;
let renderResult;

const mockSingleAffiliation = {
  displayName: 'Liz Lemon',
  email: 'liz@lemon.com',
  id: 14,
  mobilePhone: null,
  primaryPhone: '4045555555',
  role: "eAPD Federal Admin",
  secondEmail: null,
  stateId: 'md',
  status: 'requested',
  userId: '00u5mfj967KsdvBBB297',
  affiliations: [
    {
      role: "eAPD State Contractor",
      stateId: "md",
      status: "approved",
      id: 14
    }
  ]
};

const mockMultipleAffiliation = {
  displayName: 'Liz Lemon',
  email: 'liz@lemon.com',
  id: 2,
  mobilePhone: null,
  primaryPhone: '4045555555',
  role: "eAPD Federal Admin",
  secondEmail: null,
  stateId: 'md',
  status: 'requested',
  userId: '00u5mfj967KsdvBBB297',
  affiliations: [
    {
      role: "eAPD State Contractor",
      stateId: "md",
      status: "approved",
      id: 2
    },
    {
      role: "eAPD State Contractor",
      stateId: "la",
      status: "approved",
      id: 12
    }
  ]
};

const mockFedAffiliation = {
  displayName: 'Fed Admin',
  email: 'fed@admin.com',
  id: 3,
  mobilePhone: null,
  primaryPhone: '4045555555',
  role: "eAPD Federal Admin",
  secondEmail: null,
  stateId: 'fd',
  status: 'approved',
  userId: '3',
  affiliations: [
    {
      role: "eAPD Federal Admin",
      stateId: "fd",
      status: "approved",
      id: 3
    }
  ]
};

describe('<ManageAllUsersTable />', () => {
  test('shows loading when data is fetching', () => {
    props = {
      tab: 'active',
      isFetching: true,
      affiliations: [],
      actions: [],
      currentUser: { id: '123', activities: ['edit-affiliations'] }
    };
    renderResult = render(<ManageAllUsersTable {...props} />);

    const { getByText } = renderResult;
    expect(getByText('Loading...')).toBeTruthy();
  });


  test('renders correctly for users with single affiliations', () => {
    props = {
      tab: 'pending',
      affiliations: [mockSingleAffiliation],
      isFetching: false,
      actions: [],
      currentUser: { id: '123', activities: ['edit-affiliations'] }
    };
    renderResult = render(<ManageAllUsersTable {...props} />);
    expect(renderResult).toMatchSnapshot();
  });
  
  test('renders correctly for users with multiple affiliations', () => {
    props = {
      tab: 'active',
      affiliations: [mockMultipleAffiliation],
      isFetching: false,
      actions: [],
      currentUser: { id: '123', activities: ['edit-affiliations'] }
    };
    renderResult = render(<ManageAllUsersTable {...props} />);
    expect(renderResult).toMatchSnapshot();
  });
  
  test('renders correctly with both single and multi-affiliation users', () => {
    props = {
      tab: 'active',
      affiliations: [mockSingleAffiliation, mockMultipleAffiliation],
      isFetching: false,
      actions: [],
      currentUser: { id: '123', activities: ['edit-affiliations'] }
    };
    renderResult = render(<ManageAllUsersTable {...props} />);
    expect(renderResult).toMatchSnapshot();
  });

  test('renders passed in actions', () => {
    props = {
      tab: 'pending',
      affiliations: [mockSingleAffiliation],
      isFetching: false,
      actions: [
        <button type="button" key="action1">
          Take Action
        </button>
      ],
      currentUser: { id: '1234', activities: ['edit-affiliations'] }
    };
    renderResult = render(<ManageAllUsersTable {...props} />);

    const { getByText } = renderResult;
    expect(getByText('Take Action')).toBeTruthy();
  });

  test('does not render actions for ones own affiliation', () => {
    props = {
      tab: 'active',
      affiliations: [mockFedAffiliation],
      isFetching: false,
      actions: [
        <button type="button" key="action1">
          Take Action
        </button>
      ],
      currentUser: { id: '3', activities: ['edit-affiliations'] }
    };
    renderResult = render(<ManageAllUsersTable {...props} />);

    const { queryByText } = renderResult;
    expect(queryByText('Take Action')).not.toBeInTheDocument();
  });
  
  test('does not render edit action for active affiliations', () => {
    props = {
      tab: 'active',
      affiliations: [mockFedAffiliation],
      isFetching: false,
      actions: [
        <button type="button" key="action1">
          Edit Role
        </button>,
        <button type="button" key="action1">
          Revoke
        </button>
      ],
      currentUser: { id: '42', activities: ['edit-affiliations'] }
    };
    renderResult = render(<ManageAllUsersTable {...props} />);

    const { queryByText } = renderResult;
    expect(queryByText('Edit Role')).not.toBeInTheDocument();
  });


});
