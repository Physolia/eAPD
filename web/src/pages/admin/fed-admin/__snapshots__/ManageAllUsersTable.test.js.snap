// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ManageAllUsersTable /> renders correctly for users with multiple affiliations 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        class="ds-c-table ds-c-table--borderless all-users-table"
        role="table"
      >
        <thead
          role="rowgroup"
        >
          <tr
            role="row"
          >
            <th
              class="ds-c-table__cell--align-left"
              role="columnheader"
              scope="col"
            >
              Name
            </th>
            <th
              class="ds-c-table__cell--align-left"
              role="columnheader"
              scope="col"
            >
              Email
            </th>
            <th
              class="ds-c-table__cell--align-left"
              role="columnheader"
              scope="col"
            >
              Phone Number
            </th>
            <th
              class="ds-c-table__cell--align-left"
              role="columnheader"
              scope="col"
            >
              State
            </th>
            <th
              class="ds-c-table__cell--align-left"
              role="columnheader"
              scope="col"
            >
              Role
            </th>
            <th
              class="ds-c-table__cell--align-left"
              role="columnheader"
              scope="col"
            >
              Actions
            </th>
          </tr>
        </thead>
        <tbody
          role="rowgroup"
        >
          <tr
            class="all-users-table--first-row-multi"
            role="row"
          >
            <th
              class="ds-c-table__cell--align-left"
              role="rowheader"
              rowspan="2"
              style="vertical-align: top;"
            >
              Liz Lemon
            </th>
            <td
              class="ds-c-table__cell--align-left"
              role="cell"
              rowspan="2"
              style="vertical-align: top;"
            >
              liz@lemon.com
            </td>
            <td
              class="ds-c-table__cell--align-left"
              role="cell"
              rowspan="2"
              style="vertical-align: top;"
            >
              4045555555
            </td>
            <td
              class="ds-c-table__cell--align-left all-users-table--state"
              role="cell"
            >
              MD
            </td>
            <td
              class="ds-c-table__cell--align-left all-users-table--role"
              role="cell"
            >
              eAPD State Contractor
            </td>
            <td
              class="ds-c-table__cell--align-left all-users-table--actions"
              role="cell"
            >
              <div
                class="ds-u-display--flex"
                data-id="2"
                data-primary-affiliation-id="2"
                data-state="md"
              />
            </td>
          </tr>
          <tr
            class="all-users-table--row-multi"
            role="row"
          >
            <td
              class="ds-c-table__cell--align-left all-users-table--state"
              role="cell"
            >
              LA
            </td>
            <td
              class="ds-c-table__cell--align-left all-users-table--role"
              role="cell"
            >
              eAPD State Contractor
            </td>
            <td
              class="ds-c-table__cell--align-left all-users-table--actions"
              role="cell"
            >
              <div
                class="ds-u-display--flex"
                data-id="12"
                data-primary-affiliation-id="2"
                data-state="la"
              />
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      class="ds-c-table ds-c-table--borderless all-users-table"
      role="table"
    >
      <thead
        role="rowgroup"
      >
        <tr
          role="row"
        >
          <th
            class="ds-c-table__cell--align-left"
            role="columnheader"
            scope="col"
          >
            Name
          </th>
          <th
            class="ds-c-table__cell--align-left"
            role="columnheader"
            scope="col"
          >
            Email
          </th>
          <th
            class="ds-c-table__cell--align-left"
            role="columnheader"
            scope="col"
          >
            Phone Number
          </th>
          <th
            class="ds-c-table__cell--align-left"
            role="columnheader"
            scope="col"
          >
            State
          </th>
          <th
            class="ds-c-table__cell--align-left"
            role="columnheader"
            scope="col"
          >
            Role
          </th>
          <th
            class="ds-c-table__cell--align-left"
            role="columnheader"
            scope="col"
          >
            Actions
          </th>
        </tr>
      </thead>
      <tbody
        role="rowgroup"
      >
        <tr
          class="all-users-table--first-row-multi"
          role="row"
        >
          <th
            class="ds-c-table__cell--align-left"
            role="rowheader"
            rowspan="2"
            style="vertical-align: top;"
          >
            Liz Lemon
          </th>
          <td
            class="ds-c-table__cell--align-left"
            role="cell"
            rowspan="2"
            style="vertical-align: top;"
          >
            liz@lemon.com
          </td>
          <td
            class="ds-c-table__cell--align-left"
            role="cell"
            rowspan="2"
            style="vertical-align: top;"
          >
            4045555555
          </td>
          <td
            class="ds-c-table__cell--align-left all-users-table--state"
            role="cell"
          >
            MD
          </td>
          <td
            class="ds-c-table__cell--align-left all-users-table--role"
            role="cell"
          >
            eAPD State Contractor
          </td>
          <td
            class="ds-c-table__cell--align-left all-users-table--actions"
            role="cell"
          >
            <div
              class="ds-u-display--flex"
              data-id="2"
              data-primary-affiliation-id="2"
              data-state="md"
            />
          </td>
        </tr>
        <tr
          class="all-users-table--row-multi"
          role="row"
        >
          <td
            class="ds-c-table__cell--align-left all-users-table--state"
            role="cell"
          >
            LA
          </td>
          <td
            class="ds-c-table__cell--align-left all-users-table--role"
            role="cell"
          >
            eAPD State Contractor
          </td>
          <td
            class="ds-c-table__cell--align-left all-users-table--actions"
            role="cell"
          >
            <div
              class="ds-u-display--flex"
              data-id="12"
              data-primary-affiliation-id="2"
              data-state="la"
            />
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<ManageAllUsersTable /> renders correctly for users with single affiliations 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        class="ds-c-table ds-c-table--borderless all-users-table"
        role="table"
      >
        <thead
          role="rowgroup"
        >
          <tr
            role="row"
          >
            <th
              class="ds-c-table__cell--align-left"
              role="columnheader"
              scope="col"
            >
              Name
            </th>
            <th
              class="ds-c-table__cell--align-left"
              role="columnheader"
              scope="col"
            >
              Email
            </th>
            <th
              class="ds-c-table__cell--align-left"
              role="columnheader"
              scope="col"
            >
              Phone Number
            </th>
            <th
              class="ds-c-table__cell--align-left"
              role="columnheader"
              scope="col"
            >
              State
            </th>
            <th
              class="ds-c-table__cell--align-left"
              role="columnheader"
              scope="col"
            >
              Actions
            </th>
          </tr>
        </thead>
        <tbody
          role="rowgroup"
        >
          <tr
            class=""
            role="row"
          >
            <th
              class="ds-c-table__cell--align-left"
              role="rowheader"
              rowspan="1"
              style="vertical-align: top;"
            >
              Liz Lemon
            </th>
            <td
              class="ds-c-table__cell--align-left"
              role="cell"
              rowspan="1"
              style="vertical-align: top;"
            >
              liz@lemon.com
            </td>
            <td
              class="ds-c-table__cell--align-left"
              role="cell"
              rowspan="1"
              style="vertical-align: top;"
            >
              4045555555
            </td>
            <td
              class="ds-c-table__cell--align-left all-users-table--state"
              role="cell"
            >
              MD
            </td>
            <td
              class="ds-c-table__cell--align-left all-users-table--actions"
              role="cell"
            >
              <div
                class="ds-u-display--flex"
                data-id="14"
                data-primary-affiliation-id="14"
                data-state="md"
              />
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      class="ds-c-table ds-c-table--borderless all-users-table"
      role="table"
    >
      <thead
        role="rowgroup"
      >
        <tr
          role="row"
        >
          <th
            class="ds-c-table__cell--align-left"
            role="columnheader"
            scope="col"
          >
            Name
          </th>
          <th
            class="ds-c-table__cell--align-left"
            role="columnheader"
            scope="col"
          >
            Email
          </th>
          <th
            class="ds-c-table__cell--align-left"
            role="columnheader"
            scope="col"
          >
            Phone Number
          </th>
          <th
            class="ds-c-table__cell--align-left"
            role="columnheader"
            scope="col"
          >
            State
          </th>
          <th
            class="ds-c-table__cell--align-left"
            role="columnheader"
            scope="col"
          >
            Actions
          </th>
        </tr>
      </thead>
      <tbody
        role="rowgroup"
      >
        <tr
          class=""
          role="row"
        >
          <th
            class="ds-c-table__cell--align-left"
            role="rowheader"
            rowspan="1"
            style="vertical-align: top;"
          >
            Liz Lemon
          </th>
          <td
            class="ds-c-table__cell--align-left"
            role="cell"
            rowspan="1"
            style="vertical-align: top;"
          >
            liz@lemon.com
          </td>
          <td
            class="ds-c-table__cell--align-left"
            role="cell"
            rowspan="1"
            style="vertical-align: top;"
          >
            4045555555
          </td>
          <td
            class="ds-c-table__cell--align-left all-users-table--state"
            role="cell"
          >
            MD
          </td>
          <td
            class="ds-c-table__cell--align-left all-users-table--actions"
            role="cell"
          >
            <div
              class="ds-u-display--flex"
              data-id="14"
              data-primary-affiliation-id="14"
              data-state="md"
            />
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<ManageAllUsersTable /> renders correctly with both single and multi-affiliation users 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        class="ds-c-table ds-c-table--borderless all-users-table"
        role="table"
      >
        <thead
          role="rowgroup"
        >
          <tr
            role="row"
          >
            <th
              class="ds-c-table__cell--align-left"
              role="columnheader"
              scope="col"
            >
              Name
            </th>
            <th
              class="ds-c-table__cell--align-left"
              role="columnheader"
              scope="col"
            >
              Email
            </th>
            <th
              class="ds-c-table__cell--align-left"
              role="columnheader"
              scope="col"
            >
              Phone Number
            </th>
            <th
              class="ds-c-table__cell--align-left"
              role="columnheader"
              scope="col"
            >
              State
            </th>
            <th
              class="ds-c-table__cell--align-left"
              role="columnheader"
              scope="col"
            >
              Role
            </th>
            <th
              class="ds-c-table__cell--align-left"
              role="columnheader"
              scope="col"
            >
              Actions
            </th>
          </tr>
        </thead>
        <tbody
          role="rowgroup"
        >
          <tr
            class=""
            role="row"
          >
            <th
              class="ds-c-table__cell--align-left"
              role="rowheader"
              rowspan="1"
              style="vertical-align: top;"
            >
              Liz Lemon
            </th>
            <td
              class="ds-c-table__cell--align-left"
              role="cell"
              rowspan="1"
              style="vertical-align: top;"
            >
              liz@lemon.com
            </td>
            <td
              class="ds-c-table__cell--align-left"
              role="cell"
              rowspan="1"
              style="vertical-align: top;"
            >
              4045555555
            </td>
            <td
              class="ds-c-table__cell--align-left all-users-table--state"
              role="cell"
            >
              MD
            </td>
            <td
              class="ds-c-table__cell--align-left all-users-table--role"
              role="cell"
            >
              eAPD State Contractor
            </td>
            <td
              class="ds-c-table__cell--align-left all-users-table--actions"
              role="cell"
            >
              <div
                class="ds-u-display--flex"
                data-id="14"
                data-primary-affiliation-id="14"
                data-state="md"
              />
            </td>
          </tr>
          <tr
            class="all-users-table--first-row-multi"
            role="row"
          >
            <th
              class="ds-c-table__cell--align-left"
              role="rowheader"
              rowspan="2"
              style="vertical-align: top;"
            >
              Liz Lemon
            </th>
            <td
              class="ds-c-table__cell--align-left"
              role="cell"
              rowspan="2"
              style="vertical-align: top;"
            >
              liz@lemon.com
            </td>
            <td
              class="ds-c-table__cell--align-left"
              role="cell"
              rowspan="2"
              style="vertical-align: top;"
            >
              4045555555
            </td>
            <td
              class="ds-c-table__cell--align-left all-users-table--state"
              role="cell"
            >
              MD
            </td>
            <td
              class="ds-c-table__cell--align-left all-users-table--role"
              role="cell"
            >
              eAPD State Contractor
            </td>
            <td
              class="ds-c-table__cell--align-left all-users-table--actions"
              role="cell"
            >
              <div
                class="ds-u-display--flex"
                data-id="2"
                data-primary-affiliation-id="2"
                data-state="md"
              />
            </td>
          </tr>
          <tr
            class="all-users-table--row-multi"
            role="row"
          >
            <td
              class="ds-c-table__cell--align-left all-users-table--state"
              role="cell"
            >
              LA
            </td>
            <td
              class="ds-c-table__cell--align-left all-users-table--role"
              role="cell"
            >
              eAPD State Contractor
            </td>
            <td
              class="ds-c-table__cell--align-left all-users-table--actions"
              role="cell"
            >
              <div
                class="ds-u-display--flex"
                data-id="12"
                data-primary-affiliation-id="2"
                data-state="la"
              />
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      class="ds-c-table ds-c-table--borderless all-users-table"
      role="table"
    >
      <thead
        role="rowgroup"
      >
        <tr
          role="row"
        >
          <th
            class="ds-c-table__cell--align-left"
            role="columnheader"
            scope="col"
          >
            Name
          </th>
          <th
            class="ds-c-table__cell--align-left"
            role="columnheader"
            scope="col"
          >
            Email
          </th>
          <th
            class="ds-c-table__cell--align-left"
            role="columnheader"
            scope="col"
          >
            Phone Number
          </th>
          <th
            class="ds-c-table__cell--align-left"
            role="columnheader"
            scope="col"
          >
            State
          </th>
          <th
            class="ds-c-table__cell--align-left"
            role="columnheader"
            scope="col"
          >
            Role
          </th>
          <th
            class="ds-c-table__cell--align-left"
            role="columnheader"
            scope="col"
          >
            Actions
          </th>
        </tr>
      </thead>
      <tbody
        role="rowgroup"
      >
        <tr
          class=""
          role="row"
        >
          <th
            class="ds-c-table__cell--align-left"
            role="rowheader"
            rowspan="1"
            style="vertical-align: top;"
          >
            Liz Lemon
          </th>
          <td
            class="ds-c-table__cell--align-left"
            role="cell"
            rowspan="1"
            style="vertical-align: top;"
          >
            liz@lemon.com
          </td>
          <td
            class="ds-c-table__cell--align-left"
            role="cell"
            rowspan="1"
            style="vertical-align: top;"
          >
            4045555555
          </td>
          <td
            class="ds-c-table__cell--align-left all-users-table--state"
            role="cell"
          >
            MD
          </td>
          <td
            class="ds-c-table__cell--align-left all-users-table--role"
            role="cell"
          >
            eAPD State Contractor
          </td>
          <td
            class="ds-c-table__cell--align-left all-users-table--actions"
            role="cell"
          >
            <div
              class="ds-u-display--flex"
              data-id="14"
              data-primary-affiliation-id="14"
              data-state="md"
            />
          </td>
        </tr>
        <tr
          class="all-users-table--first-row-multi"
          role="row"
        >
          <th
            class="ds-c-table__cell--align-left"
            role="rowheader"
            rowspan="2"
            style="vertical-align: top;"
          >
            Liz Lemon
          </th>
          <td
            class="ds-c-table__cell--align-left"
            role="cell"
            rowspan="2"
            style="vertical-align: top;"
          >
            liz@lemon.com
          </td>
          <td
            class="ds-c-table__cell--align-left"
            role="cell"
            rowspan="2"
            style="vertical-align: top;"
          >
            4045555555
          </td>
          <td
            class="ds-c-table__cell--align-left all-users-table--state"
            role="cell"
          >
            MD
          </td>
          <td
            class="ds-c-table__cell--align-left all-users-table--role"
            role="cell"
          >
            eAPD State Contractor
          </td>
          <td
            class="ds-c-table__cell--align-left all-users-table--actions"
            role="cell"
          >
            <div
              class="ds-u-display--flex"
              data-id="2"
              data-primary-affiliation-id="2"
              data-state="md"
            />
          </td>
        </tr>
        <tr
          class="all-users-table--row-multi"
          role="row"
        >
          <td
            class="ds-c-table__cell--align-left all-users-table--state"
            role="cell"
          >
            LA
          </td>
          <td
            class="ds-c-table__cell--align-left all-users-table--role"
            role="cell"
          >
            eAPD State Contractor
          </td>
          <td
            class="ds-c-table__cell--align-left all-users-table--actions"
            role="cell"
          >
            <div
              class="ds-u-display--flex"
              data-id="12"
              data-primary-affiliation-id="2"
              data-state="la"
            />
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
